deepfence_worker/cronjobs/neo4j.go

```diff
@@ -131,9 +146,28 @@ func CleanUpDB(msg *message.Message) error {
 		return err
 	}
 
+	// host images
 	if _, err = session.Run(`
-		MATCH (n:ContainerImage)
-		WHERE n.active = false
+		MATCH (n:ContainerImage) 
+		WHERE exists((n)<-[:HOSTS]-(:Node)) 
+		AND NOT exists((n)<-[:HOSTS]-(:RegistryAccount))
+		AND n.active = false
+		AND NOT exists((n) <-[:SCANNED]-())
+		OR n.updated_at < TIMESTAMP()-$old_time_ms
+		WITH n LIMIT 100000
+		DETACH DELETE n`,
+		map[string]interface{}{
+			"old_time_ms": dbScannedResourceCleanUpTimeout.Milliseconds(),
+		}); err != nil {
+		log.Error().Msgf("Error in Clean up DB task: %v", err)
+		return err
+	}
+
+	// registry images
+	if _, err = session.Run(`
+		MATCH (n:ContainerImage) 
+		WHERE exists((n)<-[:HOSTS]-(:RegistryAccount))
+		AND n.active = false
 		AND NOT exists((n) <-[:SCANNED]-())
```

Comment:
We don't need this because `active = false` is already holding the information (shall we delete the image or not)